torchtext/datasets/conll2000chunking.py

```diff
@@ -41,12 +44,17 @@
 @_add_docstring_header(num_lines=NUM_LINES)
 @_create_dataset_directory(dataset_name=DATASET_NAME)
 @_wrap_split_argument(('train', 'test'))
-def CoNLL2000Chunking(root, split):
-    # Create a dataset specific subfolder to deal with generic download filenames
-    root = os.path.join(root, 'conll2000chunking')
-    path = os.path.join(root, split + ".txt.gz")
-    data_filename = _download_extract_validate(root, URL[split], MD5[split], path, os.path.join(root, _EXTRACTED_FILES[split]),
-                                               _EXTRACTED_FILES_MD5[split], hash_type="md5")
-    logging.info('Creating {} data'.format(split))
-    return _RawTextIterableDataset(DATASET_NAME, NUM_LINES[split],
-                                   _create_data_from_iob(data_filename, " "))
+def CoNLL2000Chunking(root: str, split: Union[Tuple[str], str]):
+    if not is_module_available("torchdata"):
+        raise ModuleNotFoundError("Package `torchdata` not found. Please install following instructions at `https://github.com/pytorch/data`")
+
+    url_dp = IterableWrapper([URL[split]])
+    cache_dp = url_dp.on_disk_cache(
+        filepath_fn=lambda x: os.path.join(root, "conll2000chunking", os.path.basename(URL[split])),
+        hash_dict={os.path.join(root, "conll2000chunking", os.path.basename(URL[split])): MD5[split]},
+        hash_type="md5"
+    )
+    cache_dp = HttpReader(cache_dp).end_caching(mode="wb", same_filepath_fn=True)
+    cache_dp = FileOpener(cache_dp, mode="b")
+    cache_dp = cache_dp.extract(file_type="gzip").readlines(decode=True)
```

Comment:
nit: can we try adding caching for extraction as well? I am trying to do for AmazonReviewPolarity here (https://github.com/pytorch/data/pull/169) though I haven't successfully done it yet. Feel free to close this PR without this change and we can follow-up later in another PR. (If you do plan to try it out, make sure to push changes only when you succeed locally otherwise it would fail the cache :) )

Reply:
I have this working locally (it seems) -- will push soon once tests pass for sure.

Reply:
Looks OK. Will push

![Tests with double caching](https://user-images.githubusercontent.com/2348806/149974377-cf75ae02-fec7-4d21-8323-662d5489c9ca.png)